### DISCUSSION TITLE
Shaprfiles do not show

### DISCUSSION THREAD

**Discussion by bclay1297** [pelicanmapping/osgearth#2651] [Support Q&A]
I am usig osgEarth 3.7. build 161. on windows 10 using visual Studio 2022. I tried loading shapefile from example data using the example code for features.  Basicly create a new osgEarth::OGRFeatureSource, set the name, set the url then mapnode get map add layer feature source.  Whe the code hits addLayer it pauses as if it is loading the data but nothing appears.  I also tried adding an entry to the Earth file but that did not show anything as well.

Unfortunately I cannot show the actual code becuase I only have internet access through my phone at this time.  
Bruce

**Comment by bclay1297** [comment]
I meant to ask are there any other dlls that have to be in the release folder to shpport shape files.  Som examples on line show setting up LineSymbols. Is that required?

**Comment by XenonofArcticus** [comment]
I recently faced this issue when the PROJ_DIR was not set and the projection support database could not be loaded.

**Comment by bclay1297** [comment]
Thanks for your reply.  I downloaded proj ver 5.0 and set the proj_dir.
Still no display

**Comment by gwaldron** [comment]
Bruce, please do this and post the output:
```
set OSGEARTH_NOTIFY_LEVEL=INFO
osgearth_features simple.earth
```
Did you install from vcpkg? Build yourself?

**Comment by bclay1297** [comment]
Glenn:
  I used vspkg to build.  The information you requested follows.

C:\Tools\EarthViewer\release>copy ..\simple.earth
Overwrite C:\Tools\EarthViewer\release\simple.earth? (Yes/No/All):
        0 file(s) copied.

C:\Tools\EarthViewer\release>osgEarth_features simple.Earth
[osgearth info] Hello, world.
Fontconfig error: Cannot load default config file: No such file: (null)
[osgearth info] [Capabilities] osgEarth Version:  3.7.0 build 161
[osgearth info] [Capabilities] GDAL Version:      3.9.1
[osgearth info] [Capabilities] OSG Version:       3.6.5
[osgearth info] [Capabilities] OSG GL3 Features:  no
[osgearth info] [Capabilities] OSG FFP Available: yes
[osgearth info] [Capabilities] CPU Cores:         16
[osgearth info] [Capabilities] GL_VENDOR:         Intel
[osgearth info] [Capabilities] GL_RENDERER:       Intel(R) UHD Graphics P630
[osgearth info] [Capabilities] GL_VERSION:        4.6.0 - Build 31.0.101.2127
[osgearth info] [Capabilities] GL CORE Profile:   no
[osgearth info] [Map] Map profile: [srs=WGS84, min=-180,-90 max=180,90 ar=2:1 vdatum=geodetic]
[osgearth info] [GDALImageLayer] "[GDALImageLayer]" [srs=WGS 84, min=-180,-90 max=180,90 ar=2:1 vdatum=geodetic] [no cache]
[osgearth info] [OGRFeatureSource] vector-data : opened OK
[osgearth warning] ASSERTION FAILURE (redraw @ FeatureModelGraph.cpp:2020) node.valid() ...
[osgearth info] [engine_rex] Activated!
[osgearth warning] ASSERTION FAILURE (createMains @ ShaderFactory.cpp:171) state.getUseVertexAttributeAliasing() ...OpenSceneGraph vertex attribute aliasing must be enabled. Consider installing the GL3RealizeOperation in your viewer.
[osgearth warning] ASSERTION FAILURE (createMains @ ShaderFactory.cpp:174) state.getUseModelViewAndProjectionUniforms() ...OpenSceneGraph matrix uniforms must be enabled. Consider installing the GL3RealizeOperation in your viewer.
VERTEX glCompileShader "oe_rex_init_model" FAILED
VERTEX Shader "oe_rex_init_model" infolog:
ERROR: 0:24: 'gl_MultiTexCoord0' : undeclared identifier
ERROR: 0:24: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:30: 'z' : vector field selection out of range


VERTEX glCompileShader "oe_rex_init_model" FAILED
VERTEX glCompileShader "oe_rex_init_model" FAILED
VERTEX Shader "oe_rex_init_model" infolog:
ERROR: 0:24: 'gl_MultiTexCoord0' : undeclared identifier
ERROR: 0:24: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:30: 'z' : vector field selection out of range


VERTEX Shader "oe_rex_normalMapVS" infolog:
ERROR: 2:33: 'gl_NormalMatrix' : undeclared identifier


VERTEX glCompileShader "oe_rex_normalMapVS" FAILED
VERTEX Shader "oe_rex_normalMapVS" infolog:
ERROR: 2:33: 'gl_NormalMatrix' : undeclared identifier


VERTEX glCompileShader "oe_rex_morph" FAILED
VERTEX Shader "oe_rex_morph" infolog:
ERROR: 2:47: 'gl_ModelViewMatrix' : undeclared identifier
ERROR: 2:66: 'gl_MultiTexCoord1' : undeclared identifier
ERROR: 2:66: 'xyz' : vector field selection out of range
ERROR: 2:67: 'constructor' : not enough data provided for construction
ERROR: 2:67: 'gl_MultiTexCoord2' : undeclared identifier
ERROR: 2:67: 'xyz' : vector field selection out of range
ERROR: 2:67: '=' :  cannot convert from 'highp float' to '3-component vector of highp float'


VERTEX Shader "oe_rex_morph" infolog:
ERROR: 2:47: 'gl_ModelViewMatrix' : undeclared identifier
ERROR: 2:66: 'gl_MultiTexCoord1' : undeclared identifier
ERROR: 2:66: 'xyz' : vector field selection out of range
ERROR: 2:67: 'constructor' : not enough data provided for construction
ERROR: 2:67: 'gl_MultiTexCoord2' : undeclared identifier
ERROR: 2:67: 'xyz' : vector field selection out of range
ERROR: 2:67: '=' :  cannot convert from 'highp float' to '3-component vector of highp float'


VERTEX Shader "oe_rex_morph" infolog:
ERROR: 2:47: 'gl_ModelViewMatrix' : undeclared identifier
ERROR: 2:66: 'gl_MultiTexCoord1' : undeclared identifier
ERROR: 2:66: 'xyz' : vector field selection out of range
ERROR: 2:67: 'constructor' : not enough data provided for construction
ERROR: 2:67: 'gl_MultiTexCoord2' : undeclared identifier
ERROR: 2:67: 'xyz' : vector field selection out of range
ERROR: 2:67: '=' :  cannot convert from 'highp float' to '3-component vector of highp float'


VERTEX glCompileShader "main(vertex)" FAILED
VERTEX Shader "main(vertex)" infolog:
ERROR: 0:48: 'gl_Vertex' : undeclared identifier
ERROR: 0:48: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:49: 'gl_Normal' : undeclared identifier
ERROR: 0:49: 'assign' :  cannot convert from 'highp float' to '3-component vector of highp float'
ERROR: 0:50: 'gl_Color' : undeclared identifier
ERROR: 0:50: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:54: 'gl_ModelViewMatrix' : undeclared identifier
ERROR: 0:55: 'gl_NormalMatrix' : undeclared identifier
ERROR: 0:62: 'gl_ProjectionMatrix' : undeclared identifier


VERTEX glCompileShader "main(vertex)" FAILED
VERTEX Shader "main(vertex)" infolog:
ERROR: 0:48: 'gl_Vertex' : undeclared identifier
ERROR: 0:48: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:49: 'gl_Normal' : undeclared identifier
ERROR: 0:49: 'assign' :  cannot convert from 'highp float' to '3-component vector of highp float'
ERROR: 0:50: 'gl_Color' : undeclared identifier
ERROR: 0:50: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:54: 'gl_ModelViewMatrix' : undeclared identifier
ERROR: 0:55: 'gl_NormalMatrix' : undeclared identifier
ERROR: 0:62: 'gl_ProjectionMatrix' : undeclared identifier


VERTEX Shader "main(vertex)" infolog:
ERROR: glLinkProgram 0000022747771870"GDALImageLayer" FAILED
glLinkProgram 0000022747770610"GDALImageLayer" FAILED
Program "GDALImageLayer" infolog:
Attached vertex shader is not compiled.

Program "GDALImageLayer" infolog:
Attached vertex shader is not compiled.

[osgearth warning] [VirtualProgram] Program will not link!
[osgearth warning] [VirtualProgram] Program will not link!
[osgearth warning] ASSERTION FAILURE (createMains @ ShaderFactory.cpp:171) state.getUseVertexAttributeAliasing() ...OpenSceneGraph vertex attribute aliasing must be enabled. Consider installing the GL3RealizeOperation in your viewer.
[osgearth warning] ASSERTION FAILURE (createMains @ ShaderFactory.cpp:174) state.getUseModelViewAndProjectionUniforms() ...OpenSceneGraph matrix uniforms must be enabled. Consider installing the GL3RealizeOperation in your viewer.
VERTEX glCompileShader "main(vertex)" FAILED
VERTEX Shader "main(vertex)" infolog:
ERROR: 0:47: 'gl_Vertex' : undeclared identifier
ERROR: 0:47: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:48: 'gl_Normal' : undeclared identifier
ERROR: 0:48: 'assign' :  cannot convert from 'highp float' to '3-component vector of highp float'
ERROR: 0:49: 'gl_Color' : undeclared identifier
ERROR: 0:49: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:52: 'gl_ModelViewMatrix' : undeclared identifier
ERROR: 0:53: 'gl_NormalMatrix' : undeclared identifier
ERROR: 0:60: 'gl_ProjectionMatrix' : undeclared identifier


VERTEX Shader "main(vertex)" infolog:
ERROR: VERTEX Shader "main(vertex)" infolog:
ERROR: 0:47: 'gl_Vertex' : undeclared identifier
ERROR: 0:47: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:48: 'gl_Normal' : undeclared identifier
ERROR: 0:48: 'assign' :  cannot convert from 'highp float' to '3-component vector of highp float'
ERROR: 0:49: 'gl_Color' : undeclared identifier
ERROR: 0:49: 'assign' :  cannot convert from 'highp float' to '4-component vector of highp float'
ERROR: 0:52: 'gl_ModelViewMatrix' : undeclared identifier
ERROR: 0:53: 'gl_NormalMatrix' : undeclared identifier
ERROR: 0:60: 'gl_ProjectionMatrix' : undeclared identifier


glLinkProgram 0000022750E9E510"Terrain image layer" FAILED
Program "Terrain image layer" infolog:
Attached vertex shader is not compiled.

[osgearth warning] [VirtualProgram] Program will not link!
glLinkProgram 0000022750E658E0"Terrain image layer" FAILED
Program "Terrain image layer" infolog:
Attached vertex shader is not compiled.

[osgearth warning] [VirtualProgram] Program will not link!
[osgearth info] Goodbye.

C:\Tools\EarthViewer\release>

**Comment by gwaldron** [comment]
Please see #2650 - I believe the answer is the same.

Include this in your application:
```c++
#include <osgEarth/GLUtils>
...
viewer.setRealizeOperation(new GL3RealizeOperation());
```

**Comment by bclay1297** [comment]
Glenn:
  I already have that line in my code.  I tried ogearth_viewer and it did
not show the shape file either.  I am wondering if some dlls are missing

**Comment by gwaldron** [comment]
You need to run `osgearth_features` from the `tests` folder in the repository. Otherwise it won't find the shapefile.

**Reply by bclay1297** [reply]
I am back home with internet access so I can continue with debugging.

I copied osgEarth_features into the test folder and added all of the required dlls.  I get a long list of errors and a bald globe.  

I don't think the errors are related to the shapefile issue but maybe you see something I don't.    When I run my app with debug set to info I get a message indicating the shapefile loads ok yet nothing shows.  I was wondering if it is a depth issue but I cannot turn on wireframe using the w.  is there a different way?


L:\SupportLibraries\osgEarth\tests>osgearth_features.exe simple.earth
Fontconfig error: Cannot load default config file: No such file: (null)
[osgearth warning] FAILED to create a terrain engine for this map
[osgearth warning] ASSERTION FAILURE (createMains @ ShaderFactory.cpp:171) state.getUseVertexAttributeAliasing() ...OpenSceneGraph vertex attribute aliasing must be enabled. Consider installing the GL3RealizeOperation in your viewer.
[osgearth warning] ASSERTION FAILURE (createMains @ ShaderFactory.cpp:174) state.getUseModelViewAndProjectionUniforms() ...OpenSceneGraph matrix uniforms must be enabled. Consider installing the GL3RealizeOperation in your viewer.
VERTEX glCompileShader "oe_LineDrawable_VS_VIEW" FAILED
VERTEX glCompileShader "oe_LineDrawable_VS_VIEW" FAILED
VERTEX Shader "oe_LineDrawable_VS_VIEW" infolog:
0(33) : error C7616: global variable gl_ModelViewMatrix is removed after version 140
0(33) : error C7616: global variable gl_Vertex is removed after version 140

VERTEX Shader "oe_LineDrawable_VS_VIEW" infolog:
0(33) : error C7616: global variable gl_ModelViewMatrix is removed after version 140
0(33) : error C7616: global variable gl_Vertex is removed after version 140

VERTEX glCompileShader "oe_LineDrawable_VS_CLIP" FAILED
VERTEX glCompileShader "oe_LineDrawable_VS_CLIP" FAILED
VERTEX glCompileShader "oe_LineDrawable_VS_CLIP" FAILED
 Shader "oe_LineDrawable_VS_CLIP" infolog:
0(41) : error C7616: global variable gl_ProjectionMatrix is removed after version 140
0(63) : error C7616: global variable gl_Vertex is removed after version 140

VERTEX glCompileShader "main(vertex)" FAILED
VERTEX glCompileShader "main(vertex)" FAILED
VERTEX glCompileShader "main(vertex)" FAILED
VERTEX Shader "main(vertex)" infolog:
0(35) : error C7616: global variable gl_Vertex is removed after version 140
0(36) : error C7616: global variable gl_Normal is removed after version 140
0(37) : error C7616: global variable gl_Color is removed after version 140
0(39) : error C7616: global variable gl_ModelViewMatrix is removed after version 140
0(40) : error C7616: global variable gl_NormalMatrix is removed after version 140
0(44) : error C7616: global variable gl_ProjectionMatrix is removed after version 140

glLinkProgram 00000216229699C0"osgEarth::LineDrawable" FAILED
Program "osgEarth::LineDrawable" infolog:
Vertex info
-----------
0(33) : error C7616: global variable gl_ModelViewMatrix is removed after version 140
0(33) : error C7616: global variable gl_Vertex is removed after version 140
(0) : error C2003: incompatible options for link
0(41) : error C7616: global variable gl_ProjectionMatrix is removed after version 140
0(63) : error C7616: global variable gl_Vertex is removed after version 140
(0) : error C2003: incompatible options for link
0(35) : error C7616: global variable gl_Vertex is removed after version 140
0(36) : error C7616: global variable gl_Normal is removed after version 140
0(37) : error C7616: global variable gl_Color is removed after version 140
0(39) : error C7616: global variable gl_ModelViewMatrix is removed after version 140
0(40) : error C7616: global variable gl_NormalMatrix is removed after version 140
0(44) : error C7616: global variable gl_ProjectionMatrix is removed after version 140
(0) : error C2003: incompatible options for link

[osgearth warning] [VirtualProgram] Program will not link!
[osgearth warning] SHADER oe_LineDrawable_VS_VIEW infolog errors:
[osgearth warning] 0:   #version 460
[osgearth warning] 1:   #extension GL_ARB_gpu_shader_int64 : enable
glLinkProgram 000002162296A8A0"osgEarth::LineDrawable" FAILED
Program "osgEarth::LineDrawable" infolog:
Vertex info
-----------
0(33) : error C7616: global variable gl_ModelViewMatrix is removed after version 140
0(33) : error C7616: global variable gl_Vertex is removed after version 140
(0) : error C2003: incompatible options for link
0(41) : error C7616: global variable gl_ProjectionMatrix is removed after version 140
0(63) : error C7616: global variable gl_Vertex is removed after version 140
(0) : error C2003: incompatible options for link
0(35) : error C7616: global variable gl_Vertex is removed after version 140
0(36) : error C7616: global variable gl_Normal is removed after version 140
0(37) : error C7616: global variable gl_Color is removed after version 140
0(39) : error C7616: global variable gl_ModelViewMatrix is removed after version 140
0(40) : error C7616: global variable gl_NormalMatrix is removed after version 140
0(44) : error C7616: global variable gl_ProjectionMatrix is removed after version 140
(0) : error C2003: incompatible options for link

[osgearth warning] 2:   #pragma vp_name GPU Lines Screen Projected Model
[osgearth warning] 3:   #pragma vp_entryPoint oe_LineDrawable_VS_VIEW
[osgearth warning] 4:   #pragma vp_location vertex_view
[osgearth warning] 5:   #pragma vp_order last
[osgearth warning] 6:   #define VP_STAGE_VERTEX
[osgearth warning] 7:   uniform vec2 oe_LineDrawable_limits;
[osgearth warning] 8:   int oe_LineDrawable_draw;
[osgearth warning] 9:   #pragma vp_varying_out flat int oe_LineDrawable_draw
[osgearth warning] 10:   // Input attributes for adjacent points
[osgearth warning] 11:   in vec3 oe_LineDrawable_prev;
[osgearth warning] 12:   in vec3 oe_LineDrawable_next;
[osgearth warning] 13:   // Shared stage globals
[osgearth warning] 14:   vec4 oe_LineDrawable_prevView;
[osgearth warning] 15:   vec4 oe_LineDrawable_nextView;
[osgearth warning] 16:   void oe_LineDrawable_VS_VIEW(inout vec4 currView)
[osgearth warning] 17:   {
[osgearth warning] 18:       // Enforce the start/end limits on the line:
[osgearth warning] 19:       oe_LineDrawable_draw = 1;
[osgearth warning] 20:       int first = int(oe_LineDrawable_limits[0]);
[osgearth warning] 21:       int last = int(oe_LineDrawable_limits[1]);
[osgearth warning] 22:       if (first >= 0)
[osgearth warning] 23:       {
[osgearth warning] 24:           if (gl_VertexID < first || (last > 0 && gl_VertexID > last))
[osgearth warning] 25:           {
[osgearth warning] 26:               oe_LineDrawable_draw = 0;
[osgearth warning] 27:           }
[osgearth warning] 28:       }
[osgearth warning] 29:       // Compute the change in the view vertex so that we can apply the same
[osgearth warning] 30:       // delta to the prev and next vectors. (An example would be if the verts
[osgearth warning] 31:       // were GPU-clamped or otherwise permuted in another shader component.)
[osgearth warning] 32:       vec4 originalView = gl_ModelViewMatrix * gl_Vertex;
[osgearth warning] 33**>     vec4 deltaView = currView - originalView;
[osgearth warning] 34:       // calculate prev/next points in post-transform view space:
[osgearth warning] 35:       oe_LineDrawable_prevView = gl_ModelViewMatrix * vec4(oe_LineDrawable_prev,1) + deltaView;
[osgearth warning] 36:       oe_LineDrawable_nextView = gl_ModelViewMatrix * vec4(oe_LineDrawable_next,1) + deltaView;
[osgearth warning] 37:   }
[osgearth warning]
[osgearth warning] 0:   #version 460
[osgearth warning] 1:   #extension GL_ARB_gpu_shader_int64 : enable
[osgearth warning] 2:   #pragma vp_name GPU Lines Screen Projected Model
[osgearth warning] 3:   #pragma vp_entryPoint oe_LineDrawable_VS_VIEW
[osgearth warning] 4:   #pragma vp_location vertex_view
[osgearth warning] 5:   #pragma vp_order last
[osgearth warning] 6:   #define VP_STAGE_VERTEX
[osgearth warning] 7:   uniform vec2 oe_LineDrawable_limits;
[osgearth warning] 8:   int oe_LineDrawable_draw;
[osgearth warning] 9:   #pragma vp_varying_out flat int oe_LineDrawable_draw
[osgearth warning] 10:   // Input attributes for adjacent points
[osgearth warning] 11:   in vec3 oe_LineDrawable_prev;
[osgearth warning] 12:   in vec3 oe_LineDrawable_next;
[osgearth warning] 13:   // Shared stage globals
[osgearth warning] 14:   vec4 oe_LineDrawable_prevView;
[osgearth warning] 15:   vec4 oe_LineDrawable_nextView;
[osgearth warning] 16:   void oe_LineDrawable_VS_VIEW(inout vec4 currView)
[osgearth warning] 17:   {
[osgearth warning] 18:       // Enforce the start/end limits on the line:
[osgearth warning] 19:       oe_LineDrawable_draw = 1;
[osgearth warning] 20:       int first = int(oe_LineDrawable_limits[0]);
[osgearth warning] 21:       int last = int(oe_LineDrawable_limits[1]);
[osgearth warning] 22:       if (first >= 0)
[osgearth warning] 23:       {
[osgearth warning] 24:           if (gl_VertexID < first || (last > 0 && gl_VertexID > last))
[osgearth warning] 25:           {
[osgearth warning] 26:               oe_LineDrawable_draw = 0;
[osgearth warning] 27:           }
[osgearth warning] 28:       }
[osgearth warning] 29:       // Compute the change in the view vertex so that we can apply the same
[osgearth warning] 30:       // delta to the prev and next vectors. (An example would be if the verts
[osgearth warning] 31:       // were GPU-clamped or otherwise permuted in another shader component.)
[osgearth warning] 32:       vec4 originalView = gl_ModelViewMatrix * gl_Vertex;
[osgearth warning] 33**>     vec4 deltaView = currView - originalView;
[osgearth warning] 34:       // calculate prev/next points in post-transform view space:
[osgearth warning] 35:       oe_LineDrawable_prevView = gl_ModelViewMatrix * vec4(oe_LineDrawable_prev,1) + deltaView;
[osgearth warning] 36:       oe_LineDrawable_nextView = gl_ModelViewMatrix * vec4(oe_LineDrawable_next,1) + deltaView;
[osgearth warning] 37:   }
[osgearth warning]
[osgearth warning] SHADER oe_LineDrawable_VS_VIEW infolog errors:
[osgearth warning] 0:   #version 460
[osgearth warning] 1:   #extension GL_ARB_gpu_shader_int64 : enable
[osgearth warning] 2:   #pragma vp_name GPU Lines Screen Projected Model
[osgearth warning] 3:   #pragma vp_entryPoint oe_LineDrawable_VS_VIEW
[osgearth warning] 4:   #pragma vp_location vertex_view
[osgearth warning] 5:   #pragma vp_order last
[osgearth warning] 6:   #define VP_STAGE_VERTEX
[osgearth warning] 7:   uniform vec2 oe_LineDrawable_limits;
[osgearth warning] 8:   int oe_LineDrawable_draw;
[osgearth warning] 9:   #pragma vp_varying_out flat int oe_LineDrawable_draw
[osgearth warning] 10:   // Input attributes for adjacent points
[osgearth warning] 11:   in vec3 oe_LineDrawable_prev;
[osgearth warning] 12:   in vec3 oe_LineDrawable_next;
[osgearth warning] 13:   // Shared stage globals
[osgearth warning] 14:   vec4 oe_LineDrawable_prevView;
[osgearth warning] 15:   vec4 oe_LineDrawable_nextView;
[osgearth warning] 16:   void oe_LineDrawable_VS_VIEW(inout vec4 currView)
[osgearth warning] 17:   {
[osgearth warning] 18:       // Enforce the start/end limits on the line:
[osgearth warning] 19:       oe_LineDrawable_draw = 1;
[osgearth warning] 20:       int first = int(oe_LineDrawable_limits[0]);
[osgearth warning] 21:       int last = int(oe_LineDrawable_limits[1]);
[osgearth warning] 22:       if (first >= 0)
[osgearth warning] 23:       {
[osgearth warning] 24:           if (gl_VertexID < first || (last > 0 && gl_VertexID > last))
[osgearth warning] 25:           {
[osgearth warning] 26:               oe_LineDrawable_draw = 0;
[osgearth warning] 27:           }
[osgearth warning] 28:       }
[osgearth warning] 29:       // Compute the change in the view vertex so that we can apply the same
[osgearth warning] 30:       // delta to the prev and next vectors. (An example would be if the verts
[osgearth warning] 31:       // were GPU-clamped or otherwise permuted in another shader component.)
[osgearth warning] 32:       vec4 originalView = gl_ModelViewMatrix * gl_Vertex;
[osgearth warning] 33**>     vec4 deltaView = currView - originalView;
[osgearth warning] 34:       // calculate prev/next points in post-transform view space:
[osgearth warning] 35:       oe_LineDrawable_prevView = gl_ModelViewMatrix * vec4(oe_LineDrawable_prev,1) + deltaView;
[osgearth warning] 36:       oe_LineDrawable_nextView = gl_ModelViewMatrix * vec4(oe_LineDrawable_next,1) + deltaView;
[osgearth warning] 37:   }
[osgearth warning]

L:\SupportLibraries\osgEarth\tests>

**Comment by gwaldron** [comment]
If you are copying DLLs around, you are going to be in a world of hurt.
Please review the vcpkg build instructions [here](https://docs.osgearth.org/en/latest/build.html#building-with-vcpkg). Specifically, look at Steps 5 & 6 where it explains how to set your PATH correctly.
